### PR TITLE
Adds automatic infrastructure provisioning guide

### PR DESCRIPTION
**Pull request by discdiver**
### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [X] This pull request includes tests or only affects documentation.
- [X] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [n/a ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed

### CODE REVIEW

File: docs/guides/deployment/push-work-pools.md [hunk 1]
```diff
@@ -19,12 +19,15 @@ Push [work pools](/concepts/work-pools/#work-pool-overview) are a special type o
...
-- Create a push work pool that sends work to Google Cloud Run, Amazon Elastic Container Service (AWS ECS) or Azure Container Instances (ACI)
+- Create a push work pool that sends work to Google Cloud Run, Amazon Elastic Container Service (AWS ECS), or Azure Container Instances (ACI)
 - Deploy a flow to that work pool
-- Execute our flow without having to run a worker or agent process to poll for flow runs
+- Execute the flow without having to run a worker or agent process to poll for flow runs
 
 ## Setup
 
+!!! note "Automatic infrastructure provisioning"
+    You can automatically provision infrastructure and create your push work pool using the `prefect work-pool create` CLI command with the `--provision-infra` flag. See the [Automatic Infrastructure Provisioning guide](/guides/deployment/auto-infra/) for more details.
```
Comment (desertaxle) [inline docs/guides/deployment/push-work-pools.md:29]: I think we could simplify this page a lot if we had `--provision-infra` as the default way to create push work pools. Could we move the info from the new page here instead?

Comment (discdiver) [reply docs/guides/deployment/push-work-pools.md:29]: > I think we could simplify this page a lot if we had --provision-infra as the default way to create push work pools. 

That's an interesting idea. I was assuming we'd want a single page to discuss all the infra provisioning options as we add more of them. That might not be a good assumption.

Comment (desertaxle) [reply docs/guides/deployment/push-work-pools.md:29]: We are scoping infrastructure provisioning to just push pools for now, so I think we can keep it all in the push pools page.

Comment (discdiver) [reply docs/guides/deployment/push-work-pools.md:29]: Ah, good to know. 
I pushed everything into the push pools page. I left the manual examples lower on the page. With tabs I think the amount of content stays manageable.

File: docs/guides/deployment/push-work-pools.md [hunk 2]
```diff
@@ -15,15 +20,232 @@ search:
...
 
-## Setup
+You can automatically provision infrastructure and create your push work pool using the `prefect work-pool create` CLI command with the `--provision-infra` flag.
+This approach greatly simplifies the setup process.
+
+Let's explore automatic infrastructure provisioning for push work pools first, and then we'll cover how to manually set up your push work pool.
+
+## Automatic infrastructure provisioning
+
+With Prefect Cloud you can provision infrastructure for use with an AWS ECS or Google Cloud Run [push work pool](/guides/deployment/push-work-pools/).
```
Comment (desertaxle) [inline docs/guides/deployment/push-work-pools.md:39]:
```suggestion
With Perfect Cloud you can provision infrastructure for use with an AWS ECS, Google Cloud Run, ACI push work pool.
```

File: docs/guides/deployment/push-work-pools.md [hunk 3]
```diff
@@ -15,15 +20,232 @@ search:
...
+    ```bash
+    prefect work-pool create --type azure-container-instance:push --provision-infra my-aci-pool
+    ```
+    </div>
+
+    Using the `--provision-infra` flag will automatically set up your default Azure account to be ready to execute flows via Azure Container Instances.
+    In your Azure account, this command will create a resource group, app registration, service account with necessary permission, generate a secret for the app registration, and create an Azure Container Instance, if they don't already exist.
+    In your Prefect workspace, this command will create an [`AzureContainerInstanceCredentials` block](https://prefecthq.github.io/prefect-azure/credentials/#prefect_azure.credentials.AzureContainerInstanceCredentials) for storing the client secret value from the generated secret.
+
+    Here's abbreviated example output from running the command:
```
Comment (desertaxle) [inline docs/guides/deployment/push-work-pools.md:124]:
```suggestion
    Here's an abbreviated example output from running the command:
```

File: docs/guides/deployment/push-work-pools.md [hunk 4]
```diff
@@ -15,15 +20,232 @@ search:
...
+    prefect work-pool create --type cloud-run:push --provision-infra my-cloud-run-pool 
+    ```
+
+    </div>
+
+    Using the `--provision-infra` flag will allow you to select a GCP project to use for your work pool and automatically configure it to be ready to execute flows via Cloud Run.
+    In your GCP project, this command will activate the Cloud Run API, create a service account, and create a key for the service account, if they don't already exist.
+    In your Prefect workspace, this command will create a [`GCPCredentials` block](https://prefecthq.github.io/prefect-gcp/credentials/) for storing the service account key.
+
+    Here's abbreviated example output from running the command:
```
Comment (desertaxle) [inline docs/guides/deployment/push-work-pools.md:176]:
```suggestion
    Here's an abbreviated example output from running the command:
```

File: docs/guides/deployment/push-work-pools.md [hunk 5]
```diff
@@ -15,15 +20,232 @@ search:
...
+Push [work pools](/concepts/work-pools/#work-pool-overview) are a special type of work pool that allows Prefect Cloud to submit flow runs for execution to serverless computing infrastructure without running a worker.
+Push work pools currently support execution in GCP Cloud Run Jobs, Azure Container Instances, and AWS ECS Tasks.
 
 In this guide you will:
 
-- Create a push work pool that sends work to Google Cloud Run, Amazon Elastic Container Service (AWS ECS) or Azure Container Instances (ACI)
+- Create a push work pool that sends work to Google Cloud Run, Amazon Elastic Container Service (AWS ECS), or Azure Container Instances (ACI)
 - Deploy a flow to that work pool
-- Execute our flow without having to run a worker or agent process to poll for flow runs
+- Execute the flow without having to run a worker or agent process to poll for flow runs
```
Comment (desertaxle) [inline docs/guides/deployment/push-work-pools.md:30]:
```suggestion
- Execute a flow without having to run a worker or agent process to poll for flow runs
```

**Review by desertaxle — APPROVED** [review]
Thanks so much for putting this together! We'll be sure to keep it up to date as we make updates to this functionality.

**Pull request merged**